twitter/twitter.php

```diff
@@ -1063,34 +1071,52 @@ function twitter_get_relation($uid, $target, $contact = [])
 	return $relation;
 }
 
-function twitter_fetch_contact($uid, $data, $create_user)
+function twitter_user_to_contact($data)
 {
 	if (empty($data->id_str)) {
 		return -1;
 	}
 
 	$avatar = twitter_fix_avatar($data->profile_image_url_https);
-	$baseurl = "https://twitter.com";
-	$url = $baseurl . "/" . $data->screen_name;
-	$addr = $data->screen_name . "@twitter.com";
+	$baseurl = 'https://twitter.com';
+	$url = $baseurl . '/' . $data->screen_name;
+	$addr = $data->screen_name . '@twitter.com';
+
+	$fields = [
+		'url'      => $url,
+		'network'  => Protocol::TWITTER,
+		'alias'    => 'twitter::' . $data->id_str,
+		'baseurl'  => $baseurl,
+		'name'     => $data->name,
+		'nick'     => $data->screen_name,
+		'addr'     => $addr,
+		'location' => $data->location,
+		'about'    => $data->description,
+		'photo'    => $avatar,
+	];
+
+	return $fields;
+}
 
-	$fields = ['url' => $url, 'network' => Protocol::TWITTER,
-		'alias' => 'twitter::' . $data->id_str,
-		'baseurl' => $baseurl, 'gsid' => GServer::getID($baseurl),
-		'name' => $data->name, 'nick' => $data->screen_name, 'addr' => $addr,
-		'location' => $data->location, 'about' => $data->description];
+function twitter_fetch_contact($uid, $data, $create_user)
+{
+	$fields = twitter_user_to_contact($data);
+
+	if (empty($fields)) {
+		return -1;
+	}
 
 	// Update the public contact
 	$pcontact = DBA::selectFirst('contact', ['id'], ['uid' => 0, 'alias' => "twitter::" . $data->id_str]);
 	if (DBA::isResult($pcontact)) {
 		$cid = $pcontact['id'];
 	} else {
-		$cid = Contact::getIdForURL($url, 0, true, $fields);
+		$cid = Contact::getIdForURL($fields['url'], 0, true, $fields);
 	}
 
 	if (!empty($cid)) {
 		DBA::update('contact', $fields, ['id' => $cid]);
-		Contact::updateAvatar($avatar, 0, $cid);
+		Contact::updateAvatar($fields['photo'], 0, $cid);
```

Comment:
The function `updateAvatar` only stores the different versions of the picture (in "photo", "thumb", "micro") when the provided url differs from the existing value in the "avatar" field. Means: We mustn't store picture data with the `DBA::update` the line before. This had to be done solely with `Contact::updateAvatar`.

Reply:
Done.